internal/service/labelstore/service.go

```diff
@@ -158,11 +158,6 @@ func (s *Service) GetOrAddGlobalRefID(l labels.Labels) uint64 {
 	s.mut.Lock()
 	defer s.mut.Unlock()
 
-	// Guard against bad input.
-	if l == nil {
-		return 0
-	}
-
```

Comment:
Are we sure this won't cause any issues? 

If we keep to be on the safe side we can just move it before the lock